File: src/TestApp/SubmitOrderConsumer.cs
using System;
using MyServiceBus;
using Microsoft.Extensions.Logging;

namespace TestApp;

class SubmitOrderConsumer :
    IConsumer<SubmitOrder>
{
    private readonly ILogger<SubmitOrderConsumer> _logger;

    public SubmitOrderConsumer(ILogger<SubmitOrderConsumer> logger)
    {
        _logger = logger;
    }

    public async Task Consume(ConsumeContext<SubmitOrder> context)
    {
        _logger.LogInformation("📨 Received SubmitOrder {OrderId} from {Message} ✅", context.Message.OrderId, context.Message.Message);

        if (DemoScenario.ShouldFaultSubmit(context.Message.Message))
        {
            _logger.LogWarning("⚠️ SubmitOrder marked as fault case");
            throw new InvalidOperationException("SubmitOrder demo fault");
        }

        var replica = Environment.GetEnvironmentVariable("HTTP_PORT") ?? Environment.MachineName;

        await context.Publish(new OrderSubmitted(context.Message.OrderId, replica));
    }
}

File: src/Java/testapp/src/main/java/com/myservicebus/testapp/SubmitOrderConsumer.java
package com.myservicebus.testapp;

import java.util.concurrent.CompletableFuture;

import com.myservicebus.ConsumeContext;
import com.myservicebus.Consumer;
import com.myservicebus.logging.Logger;
import com.myservicebus.logging.LoggerFactory;
import javax.inject.Inject;

class SubmitOrderConsumer implements Consumer<SubmitOrder> {
    private final Logger logger;

    @Inject
    public SubmitOrderConsumer(LoggerFactory loggerFactory) {
        this.logger = loggerFactory.create(SubmitOrderConsumer.class);
    }

    @Override
    public CompletableFuture<Void> consume(ConsumeContext<SubmitOrder> context) throws Exception {
        var m = context.getMessage();

        var orderId = m.getOrderId();
        var message = m.getMessage();

        if (DemoScenario.shouldFaultSubmit(message)) {
            logger.warn("⚠️ SubmitOrder marked as fault case");
            throw new IllegalStateException("SubmitOrder demo fault");
        }

        String replica = System.getenv().getOrDefault("HTTP_PORT", "unknown");
        logger.info("📨 Order id: {} (from {}) handled by {} ✅", orderId, message, replica);

        return context.publish(new OrderSubmitted(orderId, replica), context.getCancellationToken());
    }
}
